### ISSUE TITLE
Not working on cross origin

### ISSUE THREAD

**Issue by lucafaggianelli** [akfish/node-vibrant#101]
I'm using vibrant in the browser and the images are on a different server than the one serving my app, actually they're Google Places photos. I tried in both loading a string or an <img> but I always got a Cross-Origin error. My browser is Firefox.

I think cross origin images is a common scenario, how do you make it work?

```
Cross-Origin Request Blocked:
The Same Origin Policy disallows reading the remote resource at
https://maps.googleapis.com/maps/api/place/js/PhotoService.G...
(Reason: CORS header ‘Access-Control-Allow-Origin’ missing).
```

Luca

**Comment by SmartASCII** [comment]
Using cors-anywhere worked for me. All you have to do is prepend the cors-anywhere URL to the URL of your image.

`vibrantUrl= "https://cors-anywhere.herokuapp.com/" + imageUrl;`

Then use that to pass to the Vibrant builder. Note that you can't use this cors URL if you're just trying to load it into an img src or background-image, etc -- you'll need to use the direct URL for that still.

**Comment by crutchcorn** [comment]
@SmartASCII while this might work for some instances, hammering `cors-anywhere` for larger scale projects would not be ideal (nor would leaking data that way). I'd suggest allowing the headers for any `Vibrant` request to be able to be modified added as a feature to `node-vibrant`

**Comment by lucafaggianelli** [comment]
Yes the proxy could work, but as a temporary workaround, I wouldn't use it
in production. At the end I'm uploading pics to my server and issue is
solved

**Comment by Areskul** [comment]
Same issue, working on a way to fix it!

**Comment by Danistc** [comment]
Any idea on how to fix it? I'm using react-palette with images in S3 and from time to time I receive the CORS error.

**Comment by SmartASCII** [comment]
I used CORS-anywhere to ensure a solution like that would work and then rolled my own CORS proxy solution into my Node project so I could redirect locally -- been working fine for months now.

**Comment by jooohyunpark** [comment]
Any updates regarding this?

**Comment by RohitKaushal7** [comment]
anyone trying to fix this?

**Comment by Basher7** [comment]
> Using cors-anywhere worked for me. All you have to do is prepend the cors-anywhere URL to the URL of your image.
> 
> `vibrantUrl= "https://cors-anywhere.herokuapp.com/" + imageUrl;`
> 
> Then use that to pass to the Vibrant builder. Note that you can't use this cors URL if you're just trying to load it into an img src or background-image, etc -- you'll need to use the direct URL for that still.

adding cors-anywhere not working in my angular 10 project. node-vibrant unable to load images from server due to cors-policy

**Comment by yethranayeh** [comment]
> 
> 
> Using cors-anywhere worked for me. All you have to do is prepend the cors-anywhere URL to the URL of your image.
> 
> `vibrantUrl= "https://cors-anywhere.herokuapp.com/" + imageUrl;`
> 
> Then use that to pass to the Vibrant builder. Note that you can't use this cors URL if you're just trying to load it into an img src or background-image, etc -- you'll need to use the direct URL for that still.

This helped me with my APOD API project. 

Providing URL from the fetch response works when I use it on `src` attribute of the `<img>` tag, but using the same URL for vibrant results in a CORS error.

Simply prepending the link with `"https://cors-anywhere.herokuapp.com/"` helped solve my issue. Thanks!

**Comment by micahlt** [comment]
What's odd is that it seems that there is code to handle this, but it just isn't working right: https://github.com/Vibrant-Colors/node-vibrant/blob/8843d0094834d4685d1bff3abd77c6dd2daff82c/src/image/browser.ts#L49-L51

**Comment by vicvillalobos** [comment]
Cors-anywhere no longer provides this functionality in a permanent fashion, and it was never meant to be used in production builds anyway.

**Comment by Tararan** [comment]
Does anyone have a solution to this issue?
I am currently fetching data from NYT (getting books) and their books are hosted on storage.googleapis.com.

And I'm getting CORS error whenever I try using Vibrant to get colors from images.

If you have any ideas except for using cors-anywhere I'm listening

**Comment by mploux** [comment]
Same issue, any news ?

**Comment by deggertsen** [comment]
> I found a workaround, you can directly pass an Image to vibrant, and set crossOrigin to anonymous

We're doing this, but unfortunately that's not solving the problem for us either. Still looking for a solution and will report back if I find one.

**Comment by twofingerrightclick** [comment]
This is rough. Hoping someone tackles this. Its definitely very high percentage of sites where images are hosted independent of app these days. Trying out Node-Vibrant for the first time today - 3.2.1-alpha.1

**Comment by JCcastagne** [comment]
Still having this issue with a fresh NextJS + node-vibrant install.
